files/solr_ensure_instance.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash -eu
+# Ensure instance has all cores runnning fine.
+set -o pipefail
+
+if [ -z "$1" ]; then
+	echo "Cowardly refusing to reload all cores on this server. Please specify an instance as the first param."
+	exit 2
+else
+	instance="${1}"
+fi
+
+check_result="0"
+/usr/local/bin/solr_check_instance.sh "$instance" || check_result="$?"
+
+wait_time="0"
+hard_fails="0"
+until [ "$check_result" -eq "0" ]; do
+
+	wait_time=$(( wait_time + 30 ))
+	if [ "$wait_time" -gt "600" ]; then
+		echo "Solr wait timeout of 600s exceeded"
+		exit 3;
```

Comment:
any specific reason why error code 3?

Reply:
Because 2 was "bad param". Looks like the other script reuses "2" though, so should fix that one.